packages/modalConnector/src/ModalConnector.ts

```diff
@@ -0,0 +1,162 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { ethers } from 'ethers'
+import {
+  configureChains,
+  createClient,
+  getProvider,
+  getAccount,
+  getNetwork,
+  fetchSigner,
+  disconnect,
+  switchNetwork,
+  mainnet,
+  watchAccount,
+  watchNetwork,
+  GetAccountResult,
+  GetNetworkResult,
+  Chain,
+  watchProvider,
+  GetProviderResult
+} from '@wagmi/core'
+import { AbstractWeb3Connector } from '@ripio/sdk/connectors'
+import { Web3Modal } from '@web3modal/html'
+import {
+  EthereumClient,
+  modalConnectors,
+  walletConnectProvider
+} from '@web3modal/ethereum'
+import { errors } from '@ripio/sdk/types'
+
+export default class ModalConnector extends AbstractWeb3Connector {
+  web3Modal: Web3Modal
+  private _requestAccount: boolean
+  private _events: Array<CallableFunction> = []
+
+  constructor(
+    projectId: string,
+    chains: Chain[] | Chain = [mainnet],
+    requestAccount = true
+  ) {
+    super()
+    this._requestAccount = requestAccount
+
+    const _chains: Chain[] = Array.isArray(chains) ? chains : [chains]
+    // Wagmi Core Client
+    const { provider } = configureChains(_chains, [
+      walletConnectProvider({ projectId })
+    ])
+    const wagmiClient = createClient({
+      autoConnect: true,
+      connectors: modalConnectors({
+        projectId,
+        version: '1', // or "2"
```

Comment:
You should provide a more explicit comment about what this means, for example: 'This is the version of library X that we will be using. The current versions are 1 and 2, and we are setting it to 1 because... If it is changed in the future, please review compatibility.'